.circleci/config.yml

```diff
@@ -172,19 +172,65 @@ jobs:
             cd /tmp/keep-client/contracts
             gsutil -m cp * gs://${CONTRACT_DATA_BUCKET}/keep-core
 
-  build_token_dashboard_dapp:
+  publish_npm_package:
     executor: docker-node
     steps:
       - attach_workspace:
           at: /tmp/keep-client
-      - setup_remote_docker:
-          docker_layer_caching: true
       - checkout
       - run:
           name: Copy contract artifacts
-          working_directory: ~/project/solidity/dashboard
+          working_directory: ~/project/solidity
           command: |
-            ./scripts/copy-contracts.sh /tmp/keep-client/contracts
+            set -x
+            mkdir -p artifacts
+            cp -r /tmp/keep-client/contracts/* artifacts/
+            name=$(jq --raw-output .name package.json)
+            version=$(jq --raw-output .version package.json)
+            preid=$(echo $version | sed -e s/^.*-\\\([^.]*\\\).*$/\\1/)
+
+            # Find the latest published package version matching this preid.
+            latest_version=$(npm show -json "$name@^$version" version | jq --raw-output ". | sort | .[-1]")
+            latest_version=${latest_version:-$version}
+            if [ -z $latest_version ]; then
+              echo "Latest version calculation failed. Resolved info:"
+              echo "$name@$version ; preid $preid"
+              exit 1
+            fi
+
+            # Update package.json with the latest published package version matching this
+            # preid to prepare for bumping.
+            echo $(jq -M ".version=\"${latest_version}\"" package.json) > package.json
+
+            # Bump without doing any git work. Versioning is a build-time action for us.
+            # Consider including commit id? Would be +<commit id>.
```

Comment:
I like that idea.

Reply:
Let's leave this for future work? I also want to pull this into a script or orb we can reuse across repos, but want to get it out and working first.